$NetBSD: patch-configure.ac,v 1.6 2024/08/06 18:55:00 riastradh Exp $

Check for sizeof(time_t).

Leave GF_DISTRIBUTION empty in cross-builds, not set to Unspecified
which breaks things.  (Maybe we should set this in CONFIGURE_ARGS if we
can determine it.)

Defang bash 4.0 syntax in configure
(https://github.com/gluster/glusterfs/issues/3940).

--- configure.ac.orig   2022-09-20 12:24:29.568820624 +0200
+++ configure.ac        2023-03-06 22:45:21.897057433 +0100
@@ -544,6 +544,11 @@
 AC_SUBST(SIZEOF_LONG)
 AC_SUBST(SIZEOF_LONG_LONG)
 
+# size of time_t
+AC_CHECK_SIZEOF(time_t)
+SIZEOF_TIME_T=$ac_cv_sizeof_time_t
+AC_SUBST(SIZEOF_TIME_T)
+
 # YACC needs a check
 AC_PROG_YACC
 if test "x${YACC}" = "xbyacc" -o "x${YACC}" = "xyacc" -o "x${YACC}" = "x"; then
@@ -1276,8 +1281,6 @@ if test x$cross_compiling != xyes; then
     if test "x$ac_cv_file__etc_redhat_release" = "xyes"; then
         GF_DISTRIBUTION=Redhat
     fi
-else
-    GF_DISTRIBUTION=Unspecified
 fi
 
 AC_SUBST(GF_DISTRIBUTION)
@@ -1899,10 +1902,10 @@ echo "Enable Brick Mux     : $USE_BRICKM
 echo "Building with LTO    : $LTO_BUILD"
 echo
 
-# dnl Note: ${X^^} capitalization assumes bash >= 4.x
 if test "x$SANITIZER" != "xnone"; then
+        UC_SANITIZER=$(echo ${SANITIZER} | tr 'a-z' 'A-Z')
         echo "Note: since glusterfs processes are daemon processes, use"
-        echo "'export ${SANITIZER^^}_OPTIONS=log_path=/path/to/xxx.log' to collect"
+        echo "'export ${UC_SANITIZER}_OPTIONS=log_path=/path/to/xxx.log' to collect"
         echo "sanitizer output. Further details and more options can be"
         echo "found at https://github.com/google/sanitizers."
 fi
